common/buildcraft/silicon/tile/TileLaser.java

```diff
@@ -47,12 +52,14 @@
 import buildcraft.silicon.BCSiliconBlocks;
 import buildcraft.silicon.client.render.AdvDebuggerLaser;
 
-public class TileLaser extends TileBC_Neptune implements ITickable, IDebuggable {
+public class TileLaser extends TileBC_Neptune implements ITickable, IDebuggable, ILocalBlockUpdateSubscriber {
+    private List<BlockPos> targetPositions;
```

Comment:
Change to final, and use ````clear()```` rather than reallocating a new ````ArrayList```` every time.

Reply:
Oops, I wasn't thinking about the cost of memory allocation while working on a task to improve performance :man_facepalming: